providers/aws/resources/discovery.go

```diff
@@ -190,8 +190,10 @@ func Discover(runtime *plugin.Runtime, filters connection.DiscoveryFilters) (*in
 	if err != nil {
 		return nil, err
 	}
-
-	awsAccount := res.(*mqlAwsAccount)
+	var awsAccount *mqlAwsAccount
+	if res != nil {
```

Comment:
would we not need to abort here? otherwise the followup call with a nil aws account probably also would fail